src/goto-programs/goto_program.cpp

```diff
@@ -103,22 +104,55 @@ std::ostream &goto_programt::output_instruction(
   case OTHER:
     if(instruction.get_other().id() == ID_code)
     {
-      const auto &code = to_code(instruction.get_other());
+      const auto &code = instruction.get_other();
       if(code.get_statement() == ID_havoc_object)
       {
-        out << "HAVOC_OBJECT " << from_expr(ns, identifier, code.op0()) << '\n';
+        out << "HAVOC_OBJECT " << format(code.op0()) << '\n';
         break;
       }
       // fallthrough
     }
-    // fallthrough
+    out << "OTHER " << format(instruction.get_other());
+    break;
 
   case RETURN:
+    out << "RETURN " << format(instruction.return_value()) << '\n';
+    break;
+
   case DECL:
+    out << "DECL " << format(instruction.decl_symbol()) << " : "
+        << format(instruction.decl_symbol().type()) << '\n';
+    break;
+
   case DEAD:
+    out << "DEAD " << format(instruction.dead_symbol()) << '\n';
+    break;
+
   case FUNCTION_CALL:
+    out << "CALL ";
+    {
+      auto &call = instruction.get_function_call();
```

Comment:
Is this the only place where we print `codet`? It might be worth moving the formatting into a `util/format_code.h` (not in this PR) otherwise.

Reply:
The idea is to get rid of the use of `codet` in goto functions altogether; function calls are in fact the only instruction that doesn't have an alternative API yet.